src/gfx/2d/2d.ts

```diff
@@ -197,6 +198,7 @@ module Shumway.GFX.Canvas2D {
 
     set (state: RenderState) {
       this.clip.set(state.clip);
+      this.clipPath = state.clipPath;
```

Comment:
After discussing this with @tobytailor it turns out this is in fact correct. However, we should add a comment as to why that is.